specification/k3po.control/SPEC.md

```diff
@@ -0,0 +1,195 @@
+# K3po Control Protocol
+
+## Abstract
+
+This document defines the K3po control protocol used to communicate with a remote K3po server.
+
+## Requirements
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY",
+and "OPTIONAL" in this document are to be interpreted as described in [RFC2119](https://tools.ietf.org/html/rfc2119).
+
+## The Protocol
+
+The protocol is made up of Commands and Events.  Commands are actions that can be triggered by an external entity.
+Events are triggered by observed actions or state changes and are sent back to the external entity.  The protocol 
+is formatted by
+
+TODO normalize
+
+```
+Command / Event Name\n
+headers
+empty line
+content
+
+```
+
+headers are key value pairs separated by a ":" and delimited by a new line "\n"
+
+If the "content-length" header is specified, then optional content of the specified length can be added.
+
+Headers that are not understood / processed by a command can be ignored.
```

Comment:
Confirmed that our implementation does have separate catch and exception for content.  See https://github.com/k3po/k3po/blob/develop/driver/src/main/java/org/kaazing/k3po/driver/internal/control/handler/ControlDecoder.java line 316.  

If we specify that unrecognized content should be ignored then we need to change this implementation.  Which is probably better for future compatibility